src/Storefront/Resources/app/storefront/src/plugin/wishlist/wishlist-cookie-offcanvas.plugin.js

```diff
@@ -0,0 +1,137 @@
+import Plugin from 'src/plugin-system/plugin.class';
+import CookieStorageHelper from 'src/helper/storage/cookie-storage.helper';
+import AjaxOffCanvas from 'src/plugin/offcanvas/ajax-offcanvas.plugin';
+
+/**
+ * @package checkout
+ */
+export default class WishlistCookieOffcanvasPlugin extends Plugin {
+    static options = {
+        cookieName: 'wishlist-enabled',
+        cookieLifetime: 30,
+    };
+
+    init() {
+        this._registerEvents();
+    }
+
+    /**
+     * @returns {boolean}
+     */
+    static hasConsent() {
+        if (!CookieStorageHelper.isSupported()) {
+            return false;
+        }
+        return CookieStorageHelper.getItem(this.options.cookieName) === '1';
+    }
+
+    /**
+     * @param {string|number} productId
+     * @param {Function} onConsent
+     */
+    static requestConsent(productId, onConsent) {
+        const url = window.router['frontend.wishlist.cookie.offcanvas'];
+
+        AjaxOffCanvas.open(url, false, () => {
+            window.PluginManager.initializePlugins();
+
+            const offcanvas = document.querySelector('.offcanvas');
+            if (!offcanvas) {
+                return;
+            }
+
+            const plugin = window.PluginManager
+                .getPluginInstanceFromElement(offcanvas, 'WishlistCookieOffcanvas');
+
+            if (plugin) {
+                plugin.options.productId = productId;
+                plugin.$emitter.subscribe('WishlistCookie/onAccept', onConsent);
+            }
+        }, 'left');
+    }
+
+    /**
+     * @private
+     */
+    _registerEvents() {
+        const acceptBtn = this.el.querySelector('.js-wishlist-cookie-accept');

```

Comment:
Please make these 4 classes configurable via the `options` object, otherwise looking really good to me 🚀